tap_zuora/__init__.py

```diff
@@ -47,18 +42,17 @@ def validate_state(config: dict, catalog: Catalog, state: dict) -> dict:
         else:
             LOGGER.info("No bookmarks found")
             state["bookmarks"] = {}
-
     if "current_stream" not in state:
         LOGGER.info("Current stream not found")
         state["current_stream"] = None
 
     for stream in catalog.streams:
-        if not stream_is_selected(metadata.to_map(stream.metadata)):
+        if not stream.is_selected():
             if state["current_stream"] == stream.tap_stream_id:
                 state["current_stream"] = None
             continue
 
-        if stream.tap_stream_id not in state["bookmarks"]:
+        if stream.tap_stream_id not in state["bookmarks"] or not state["bookmarks"][stream.tap_stream_id].get('version'):
```

Comment:
It's possible that `None` could be a valid value for version in some situations. This API is stateful, so this might be an impactful change. I can't tell, but it's worrisome.